src/topology/perfect.lean

```diff
@@ -0,0 +1,279 @@
+theorem frequently_nhds_nhds {x : α} {p : α → α → Prop} :
+  (∃ᶠ y in 𝓝 x, ∃ᶠ z in 𝓝 y, p x z) → (∃ᶠ z in 𝓝 x, p x z) :=
+begin
+  rw [(nhds_basis_opens _).frequently_iff, (nhds_basis_opens _).frequently_iff],
+  rintros h U ⟨xU,Uop⟩,
+  rcases h U ⟨xU,Uop⟩ with ⟨y,yU,hy⟩,
+  rw (nhds_basis_opens _).frequently_iff at hy,
+  rcases hy U ⟨yU,Uop⟩ with ⟨z,zU,hz⟩,
+  use [z,zU,hz],
+end
+
+/-- In a T1 space, being preperfect is equivalent to having perfect closure.-/
+theorem preperfect_iff_perfect_closure [t1_space α] {C : set α} :
+  preperfect C ↔ perfect (closure C) :=
+begin
+  split; intro h, {exact h.perfect_closure},
+  intros x xC,
+  have H := h.acc _ (subset_closure xC),
+  rw acc_pt_iff_frequently at *,
+  have : ∀ y , y ≠ x ∧ y ∈ closure C → ∃ᶠ z in 𝓝 y, z ≠ x ∧ z ∈ C,
+  { rintros y ⟨hyx, yC⟩,
+    simp only [← mem_compl_singleton_iff, @and_comm _ (_ ∈ C) , ← frequently_nhds_within_iff,
+      hyx.nhds_within_compl_singleton, ← mem_closure_iff_frequently],
+    exact yC, },
+  exact frequently_nhds_nhds (H.mono this),
+end
+
+theorem perfect.closure_nhd_inter {C U: set α} (hC : perfect C) (x : α) (xC : x ∈ C) (xU : x ∈ U)
+  (Uop : is_open U) : perf_nonempty (closure (U ∩ C)) :=
+begin
+  split,
+  { apply preperfect.perfect_closure,
+    exact (hC.acc).open_inter Uop, },
+  apply nonempty.mono subset_closure,
+  exact ⟨x,⟨xU,xC⟩⟩,
+end
+
+/-- Given a perfect nonempty set in a T2.5 space, we can find two disjoint perfect subsets
+This is the main inductive step in the proof of the Cantor-Bendixson Theorem-/
+lemma perf_nonempty.splitting [t2_5_space α] {C : set α} (hC : perf_nonempty C) :
+  ∃ C₀ C₁ : set α, (perf_nonempty C₀ ∧ C₀ ⊆ C) ∧ (perf_nonempty C₁ ∧ C₁ ⊆ C) ∧ disjoint C₀ C₁ :=
+begin
+  cases hC.nonempty with y yC,
+  obtain ⟨x, xC, hxy⟩ : ∃ x ∈ C, x ≠ y,
+  { have := hC.acc _ yC,
+    rw acc_pt_iff_nhds at this,
+    rcases this univ (univ_mem) with ⟨x,xC,hxy⟩,
+    exact ⟨x,xC.2,hxy⟩, },
+  obtain ⟨U, xU, Uop, V, yV, Vop, hUV⟩ := exists_open_nhds_disjoint_closure hxy,
+  use [closure (U ∩ C), closure (V ∩ C)],
+  split,
+  { split, { apply hC.closure_nhd_inter x xC xU Uop, },
+    rw hC.closed.closure_subset_iff,
+    apply inter_subset_right, },
+  split,
+  { split, { apply hC.closure_nhd_inter y yC yV Vop, },
+    rw hC.closed.closure_subset_iff,
+    apply inter_subset_right, },
+  apply disjoint.mono _ _ hUV; apply closure_mono; apply inter_subset_left,
+end
+
+section kernel
+
+/-- The Cantor-Bendixson Theorem: Any closed subset of a second countable space
+can be written as the union of a countable set and a perfect set.-/
+theorem ctble_union_perfect_of_closed [second_countable_topology α] {C : set α}
+  (hclosed : is_closed C) : ∃ V D : set α, (V.countable) ∧ (perfect D) ∧ (C = V ∪ D) :=
+begin
+  have := topological_space.exists_countable_basis α,
+  rcases this with ⟨b,bct,bnontrivial,bbasis⟩,
+  let v := {U ∈ b | (U ∩ C).countable},
+  let V := ⋃ U ∈ v, U,
+  let D := C ∩ Vᶜ, --C \ V did not work
```

Comment:
It does work with `C \ V` (note that it is not a regular `\`). You just have to replace `is_closed.inter hclosed` by `hclosed.sdiff` and `inter_union_compl` by `inter_diff`

Reply:
changed, thanks!